### PR TITLE
The Elephant in the Room

### PR DESCRIPTION
**Pull request by AntoineTheb**

### CODE REVIEW

File: deep-learning/_posts/2018-09-27-the_elephant_in_the_room.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+### Contextual reasoning
+It is possible that network end up doing some contextual reasoning even if it is not explicitely learned, explaining why matching two objects that were never shown together during training (an elephant and a book, for example) might cause catasprophic failures.
+
+### Non-maximal suppression
+Assume an object A that is no longer being detected because it is now (partially) occluded by an object T. An object B, which was partially occluded by A and not detected, might now be detected. This chain-reaction could cause non-local effects.
+
+### Feature inference:
+![](/deep-learning/images/the_elephant_in_the_room/cat.png)
+
+It is possible that features outside the object's ROI might influence the object's detection. This could help object detection in cases where the object is partially occluded, but also hinder object detection. Pixels in the ROI which are not the object (like backgroun) could also influence detection. The authors suggest that this is the main cause of problems.
```
Comment (Dref360) [inline deep-learning/_posts/2018-09-27-the_elephant_in_the_room.md]: background*

File: deep-learning/_posts/2018-09-27-the_elephant_in_the_room.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+The first test was to extract an object (a elephant, wink) from another image and transplant it at various locations in another image (a living room, wink) on which the detector was trained (Faster-RCNN with NASNet backbone). Results:
+
+![](/deep-learning/images/the_elephant_in_the_room/litteral-elephant.png)
+![](/deep-learning/images/the_elephant_in_the_room/polar-bear.png)
+![](/deep-learning/images/the_elephant_in_the_room/sandwich.png)
+![](/deep-learning/images/the_elephant_in_the_room/bus.png)
+![](/deep-learning/images/the_elephant_in_the_room/horse.png)
+![](/deep-learning/images/the_elephant_in_the_room/surfboard.png)
+
```
Comment (Dref360) [inline deep-learning/_posts/2018-09-27-the_elephant_in_the_room.md]: Add a --- to separate those section, not easy to see the text between the images

**Comment by AntoineTheb** [comment]
@Dref360 Fixed, thanks !

File: deep-learning/_posts/2018-09-27-the_elephant_in_the_room.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+---
+
+Anectodically, they also tested Google OCR and it also broke:
+
+![](/deep-learning/images/the_elephant_in_the_room/google.png)
+
+## The results
+![](/deep-learning/images/the_elephant_in_the_room/table.png)
+where 'tau' is the threshold of accepted difference between bounding boxes
```
Comment (Dref360) [inline deep-learning/_posts/2018-09-27-the_elephant_in_the_room.md]: add a blank line after the image, won't render right.

Comment (AntoineTheb) [reply deep-learning/_posts/2018-09-27-the_elephant_in_the_room.md]: Woops, thanks

**Comment by AntoineTheb** [comment]
@Dref360 plz approve

**Pull request merged**